### PR TITLE
Add workflow for publishing MSIX package

### PR DESCRIPTION
**Pull request by yaakovschectman**
## Description

Add an example workflow for packaging an MSIX and publishing it to the Microsoft Store.

Adjacent to https://github.com/flutter/website/pull/8539

### CODE REVIEW

File: .github/workflows/dart.yml [hunk 1]
```diff
@@ -0,0 +1,45 @@
+name: Dart
+
+on: workflow_dispatch
+
+jobs:
+  build:
+    runs-on: windows-latest
+    environment: WindowsEnv
```
Comment (loic-sharma) [inline .github/workflows/dart.yml:8]: What does this do?

Comment (yaakovschectman) [reply .github/workflows/dart.yml:8]: This specifies an environment containing the secrets used by the workflow.

File: .github/workflows/dart.yml [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+        run: dart analyze
+
+      # Your project will need to have tests in test/ and a dependency on
+      # package:test for this step to succeed. Note that Flutter projects will
+      # want to change this to 'flutter test'.
+      - name: Run tests
+        run: dart test
+      
+      - name: Run flutter doctor
+        run: flutter doctor -v
```
Comment (loic-sharma) [inline .github/workflows/dart.yml:39]: Is this necessary to publish? Or can we cut it? The Play Store example does not run tests/analysis/doctor

Comment (yaakovschectman) [reply .github/workflows/dart.yml:39]: It is not necessary. We can remove it.

File: .github/workflows/dart.yml [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+      
+      - name: MS package
```
Comment (loic-sharma) [inline .github/workflows/dart.yml:41]:
```suggestion
      - name: Create MSIX package
```

File: .github/workflows/dart.yml [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+        run: msstore package .
+      
+      - name: MS publish
```
Comment (loic-sharma) [inline .github/workflows/dart.yml:27]:
```suggestion
      - name: Publish MSIX to the Microsoft Store

```

File: .github/workflows/dart.yml (hunk 1 above)
Comment (loic-sharma) [reply .github/workflows/dart.yml:8]: We should make this example as simple as possible. Can we cut this if we use repository secrets instead?

File: .github/workflows/dart.yml [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+        run: msstore reconfigure --tenantId ${{ secrets.AZURE_AD_TENANT_ID }} --clientId ${{ secrets.AZURE_AD_ClIENT_ID }} --clientSecret ${{ secrets.AZURE_AD_CLIENT_SECRET }} --sellerId ${{ secrets.SELLER_ID }}
+
+      - name: Install dependencies
+        run: dart pub get
+      
+      - name: Create MSIX package
+        run: msstore package .
+      
+      - name: Publish MSIX to the Microsoft Store
+
```
Comment (loic-sharma) [inline .github/workflows/dart.yml:27]:
```suggestion

```

File: .github/workflows/dart.yml
Comment (guidezpl) [inline .github/workflows/dart.yml:1]: please rename `release_deploy_windows.yml` to match other release workflows

**Review by guidezpl — CHANGES_REQUESTED** [review]

**Review by loic-sharma — APPROVED** [review]
Looks good to me but please get approval from someone else who is more familiar with the gallery

**Review by guidezpl — CHANGES_REQUESTED** [review]
LGTM but could you add a blurb about this to the releasing instructions in the README, linking to the Windows store listing?

**Comment by yaakovschectman** [comment]
@guidezpl What Windows store listing are you referring to?

**Comment by loic-sharma** [comment]
@yaakovschectman I think the idea is to update the README's [supported platforms section](https://github.com/flutter/gallery#supported-platforms) to link to Gallery app on the Microsoft Store and the [releasing section](https://github.com/flutter/gallery#releasing) to explain how to deploy the Gallery app to the Microsoft Store.

**Review by guidezpl — APPROVED** [review]
```
  _    ___ _____ __  __ 
 | |  / __|_   _|  \/  |
 | |_| (_ | | | | |\/| |
 |____\___| |_| |_|  |_|
                        
```

**Comment by guidezpl** [comment]
You should be able to merge, right?

**Pull request merged**